app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java

```diff
@@ -265,15 +265,20 @@ public void notificationSetter(Context context) {
         }
     }
 
-    private void  ArmTimer(Context context, int time) {
+    private void  ArmTimer() {
         Log.e(TAG, "ArmTimer called");
-        if(ActiveBgAlert.getOnly() != null) {
-            Calendar calendar = Calendar.getInstance();
-            AlarmManager alarm = (AlarmManager) getSystemService(ALARM_SERVICE);
-            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.KITKAT) {
-                alarm.setExact(AlarmManager.RTC_WAKEUP, calendar.getTimeInMillis() + time, PendingIntent.getService(this, 0, new Intent(this, Notifications.class), 0));
-            } else {
-                alarm.set(AlarmManager.RTC_WAKEUP, calendar.getTimeInMillis() + time, PendingIntent.getService(this, 0, new Intent(this, Notifications.class), 0));
+        ActiveBgAlert activeBgAlert = ActiveBgAlert.getOnly();
+        if(activeBgAlert != null ) {
+            AlertType alert = AlertType.get_alert(activeBgAlert.alert_uuid);
+            if(alert != null) {
+                int time = alert.minutes_between;
+                Calendar calendar = Calendar.getInstance();
+                AlarmManager alarm = (AlarmManager) getSystemService(ALARM_SERVICE);
+                if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.KITKAT) {
+                    alarm.setExact(AlarmManager.RTC_WAKEUP, calendar.getTimeInMillis() + time, PendingIntent.getService(this, 0, new Intent(this, Notifications.class), 0));
+                } else {
+                    alarm.set(AlarmManager.RTC_WAKEUP, calendar.getTimeInMillis() + time, PendingIntent.getService(this, 0, new Intent(this, Notifications.class), 0));
```

Comment:
Do you know what the issue is?
will you have a fix soon?
Should I work on this branch, to make my fix?

Thanks
Tzachi

On Wed, Jun 10, 2015 at 11:27 PM, Stephen Black <notifications@github.com>
wrote:

> In
> app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java
> <https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/49#discussion_r32162269>
> :
>
> > -            AlarmManager alarm = (AlarmManager) getSystemService(ALARM_SERVICE);
> > -            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.KITKAT) {
> > -                alarm.setExact(AlarmManager.RTC_WAKEUP, calendar.getTimeInMillis() + time, PendingIntent.getService(this, 0, new Intent(this, Notifications.class), 0));
> > -            } else {
> > -                alarm.set(AlarmManager.RTC_WAKEUP, calendar.getTimeInMillis() + time, PendingIntent.getService(this, 0, new Intent(this, Notifications.class), 0));
> > +        ActiveBgAlert activeBgAlert = ActiveBgAlert.getOnly();
> > +        if(activeBgAlert != null ) {
> > +            AlertType alert = AlertType.get_alert(activeBgAlert.alert_uuid);
> > +            if(alert != null) {
> > +                int time = alert.minutes_between;
> > +                Calendar calendar = Calendar.getInstance();
> > +                AlarmManager alarm = (AlarmManager) getSystemService(ALARM_SERVICE);
> > +                if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.KITKAT) {
> > +                    alarm.setExact(AlarmManager.RTC_WAKEUP, calendar.getTimeInMillis() + time, PendingIntent.getService(this, 0, new Intent(this, Notifications.class), 0));
> > +                } else {
> > +                    alarm.set(AlarmManager.RTC_WAKEUP, calendar.getTimeInMillis() + time, PendingIntent.getService(this, 0, new Intent(this, Notifications.class), 0));
>
> note for self: set that badboy * 1000
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/49/files#r32162269>
> .
>